ios/AppLovinMAXAdView.m

```diff
@@ -0,0 +1,204 @@
+    }
+    
+    if ( [@"banner" isEqualToString: adFormat] )
+    {
+        _adFormat = DEVICE_SPECIFIC_ADVIEW_AD_FORMAT;
+    }
+    else if ( [@"mrec" isEqualToString: adFormat] )
+    {
+        _adFormat = MAAdFormat.mrec;
+    }
+    else
+    {
+        [[AppLovinMAX shared] log: @"Attempting to set an invalid ad format of \"%@\" for %@", adFormat, self.adUnitId];
+        return;
+    }
+    
+    [self attachAdView];
+}  
+
+- (void)setPlacement:(NSString *)placement
+{
+    _placement = placement;
+    
+    if ( self.adView )
+    {
+        self.adView.placement = placement;
+    }
+}
+
+- (void)setCustomData:(NSString *)customData
+{
+    _customData = customData;
+    
+    if ( self.adView )
+    {
+        self.adView.customData = customData;
+    }
+}
+
+- (void)setAdaptiveBannerEnabled:(BOOL)adaptiveBannerEnabled
+{
+    _adaptiveBannerEnabled = adaptiveBannerEnabled;
+    
+    if ( self.adView )
+    {
+        [self.adView setExtraParameterForKey: @"adaptive_banner" value: adaptiveBannerEnabled ? @"true" : @"false"];
+    }
+}
+
+- (void)setAutoRefresh:(BOOL)autoRefresh
+{
+    _autoRefresh = autoRefresh;
+    
+    if ( self.adView )
+    {
+        if ( autoRefresh )
+        {
+            [self.adView startAutoRefresh];
+        }
+        else
+        {
+            [self.adView stopAutoRefresh];
+        }
+    }
+}
+
+- (void)attachAdView
+{
+    if ( ![self.adUnitId al_isValidString] )
+    {
+        [[AppLovinMAX shared] log: @"Attempting to attach MAAdView without Ad Unit ID"];
+        return;
+    }
+    
+    if ( !self.adFormat )
+    {
+        [[AppLovinMAX shared] log: @"Attempting to attach MAAdView without ad format"];
+        return;
+    }
+    
+    // Re-assign in case of race condition
+    NSString *adUnitId = self.adUnitId;
+    MAAdFormat *adFormat = self.adFormat;
+    
+    // Run after 0.25 sec delay to allow all properties to set
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t) (NSEC_PER_SEC/4)), dispatch_get_main_queue(), ^{
+        if ( self.adView )
+        {
+            [[AppLovinMAX shared] log: @"Attempting to re-attach with existing MAAdView: %@", self.adView];
+            return;
+        }
+        [[AppLovinMAX shared] log: @"Attaching MAAdView..."];
+        self.adView = [[MAAdView alloc] initWithAdUnitIdentifier: adUnitId
+                                                        adFormat: adFormat
+                                                             sdk: AppLovinMAX.shared.sdk];
+        self.adView.frame = (CGRect) { CGPointZero, adFormat.size };
+        self.adView.delegate = AppLovinMAX.shared; // Go through core class for callback forwarding to React Native layer
+        self.adView.revenueDelegate = AppLovinMAX.shared;
+        if ( self.placement )
+        {
+            self.adView.placement = self.placement;
+        }
+        if ( self.customData )
+        {
+            self.adView.customData = self.customData;
+        }
```

Comment:
1. nit - match Android formatting
```suggestion
        self.adView.revenueDelegate = AppLovinMAX.shared;
        
        if ( self.placement )
        {
            self.adView.placement = self.placement;
        }
        
        if ( self.customData )
        {
            self.adView.customData = self.customData;
        }
```

Reply:
2. Is it possible that the value is updated in the RN DOM attribute to a value of `nil`? For instance - the pub sets a legit value of `"foo"` for placement or custom data, then perhaps after X seconds later updates it to `""` or removes the attribute (not sure if possible). In that case - I'm afraid that we will not update the `adView`'s placement or customData since we only set for non-nil.

Which begs the question - is there a reason why we only set for non-nil? Why not just set it all the time? `MAAdView` / `MaxAdView` defaults to `nil` / `null` anyways.